data_adapter_oemof/build_datapackage.py

```diff
@@ -1,92 +1,185 @@
 import dataclasses
-import os
-from collections import defaultdict
+import warnings
 
 import pandas as pd
 
-import data_adapter.preprocessing as preprocessing
+from data_adapter import core
+from data_adapter.preprocessing import Adapter
 from data_adapter_oemof.adapters import TYPE_MAP
 from data_adapter_oemof.mappings import PROCESS_TYPE_MAP
 from oemof.tabular.datapackage.building import infer_metadata
 
 
+def refactor_timeseries(timeseries: pd.DataFrame):
+    """
+    Takes timeseries in parameter-model format (as a single line entry) And
+    turns into Tabular matching format with index as timeseries timestamps
+    and columns containing data
+
+    :return: pd.DataFrame:
+
+    """
+
+    # Combine all time series into one DataFrame
+    df_timeseries = pd.DataFrame()
+    for (start, end, freq, region), df in timeseries.groupby(
+        ["timeindex_start", "timeindex_stop", "timeindex_resolution", "region"]
+    ):
+        timeindex = pd.date_range(start=start, end=end, freq=pd.Timedelta(freq))
+
+        # Get column names of timeseries only
+        ts_columns = set(df.columns).difference(core.TIMESERIES_COLUMNS.keys())
+
+        # Iterate over columns in case there are multiple timeseries
+        for profile_name in ts_columns:
+            profile_column = df[profile_name].dropna()
+            profile_column = profile_column.explode().to_frame()
+            profile_column["timeindex"] = profile_column.groupby(level=0).cumcount()
+            profile_column = profile_column.reset_index().pivot(
+                index="timeindex", columns="index", values=profile_name
+            )
```

Comment:
What's happening here?
Hard to understand

Reply:
I agree, it's complicated but we didn't find a way to unnest and reformat multiple arrays at once into columns while keeping timeindex order. I added further comments to explain the process.